partitura/performance.py

```diff
@@ -280,6 +284,96 @@ def adjust_offsets_w_sustain(
         note["sound_off"] = offset
 
 
+class PerformedNote(dict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self["id"] = self.get("id", None)
+        self["pitch"] = self.get("pitch", self["midi_pitch"])
+        self["note_on"] = self.get("note_on", -1)
+        self["note_off"] = self.get("note_off", -1)
+        self["sound_off"] = self.get("sound_off", self["note_off"])
+        self["track"] = self.get("track", 0)
+        self["channel"] = self.get("channel", 1)
+        self["velocity"] = self.get("velocity", 60)
+        self.validate_values()
+        self._accepted_keys = ["id", "pitch", "note_on", "note_off", "velocity", "track", "channel", "sound_off"]
+        self.__setitem__ = self._setitem_new
+
+    def __repr__(self):
+        return f"PerformedNote: {self['id']}"
+
+    def __str__(self):
+        return f"PerformedNote: {self['id']}"
+
+    def __eq__(self, other):
+        return self["id"] == other["id"]
+
+    def __hash__(self):
+        return hash(self["id"])
+
+    def __lt__(self, other):
+        return self["note_on"] < other["note_on"]
+
+    def __le__(self, other):
+        return self["note_on"] <= other["note_on"]
+
+    def __gt__(self, other):
+        return self["note_on"] > other["note_on"]
+
+    def __ge__(self, other):
+        return self["note_on"] >= other["note_on"]
+
+    def __getitem__(self, key):
+        return self.get(key, None)
+
+    def _setitem_new(self, key, value):
+        if key not in self._accepted_keys:
+            raise KeyError(f"Key {key} not in PerformedNote")
+        elif key == "note_off":
+            # Verify that the note_off is after the note_on
+            if value < self["note_on"]:
+                raise ValueError(f"note_off must be after note_on")
+            self["sound_off"] = value if self["sound_off"] < value else self["sound_off"]
+            self["note_off"] = value
+        elif key == "note_on":
+            # Verify that the note_on is before the note_off
+            if value > self["note_off"]:
```

Comment:
This would actually be notes that weren't actually played, or risk of having a non-note off property. I don't know I was thinking about it when writing the verifier but.... what do you think?

Reply:
Apparently, there are some MIDI files when 0 note length happen (though I would forbid it, but that's it in the current settings). So if you raise an exception, this could block the loading of such MIDIs. 

For this reason, we should allow equality, which means your actual function is correct.